service/qos/service_level_controller.hh

```diff
@@ -235,12 +235,12 @@ public:
      * update_both_cache_levels::yes - updates both levels of the cache
      * update_both_cache_levels::no  - update only effective service levels cache
      */
-    future<> update_cache(update_both_cache_levels update_both_cache_levels = update_both_cache_levels::yes);
+    future<> update_cache(update_both_cache_levels update_both_cache_levels = update_both_cache_levels::yes, qos::query_context ctx = qos::query_context::unspecified);
```

Comment:
Similarly here, `service_level_controller::update_cache()` is only called in group0 context. Legacy service levels v1 were calling `service_level_controller::update_service_levels_cache()` from the update loop.

Reply:
Small correction: I think you actually meant https://github.com/scylladb/scylladb/pull/21773